src/executePrettier/executePrettierOnBufferRange.js

```diff
@@ -12,60 +11,80 @@ const buildPrettierStylelintOptions = require('./buildPrettierStylelintOptions')
 const buildPrettierOptions = require('./buildPrettierOptions');
 const handleError = require('./handleError');
 
-const executePrettier = (editor: TextEditor, text: string) =>
-  getPrettierInstance(editor).format(text, buildPrettierOptions(editor));
+const executePrettierWithCursor = (
+  editor: TextEditor,
+  text: string,
+  cursorOffset: number,
+): Prettier$CursorResult =>
+  getPrettierInstance(editor).formatWithCursor(text, {
+    ...buildPrettierOptions(editor),
+    cursorOffset,
+  });
 
-const executePrettierEslint = (editor: TextEditor, text: string) =>
+const executePrettierEslint = (editor: TextEditor, text: string): string =>
   allowUnsafeNewFunction(() => prettierEslint(buildPrettierEslintOptions(editor, text)));
 
-const executePrettierStylelint = (editor: TextEditor, text: string) =>
+const executePrettierStylelint = (editor: TextEditor, text: string): string =>
   prettierStylelint.format(buildPrettierStylelintOptions(editor, text));
 
-const executePrettierOrIntegration = async (editor: TextEditor, text: string) => {
-  try {
-    let formatted;
-    if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
-      formatted = await executePrettierStylelint(editor, text);
-    } else if (shouldUseEslint()) {
-      formatted = executePrettierEslint(editor, text);
-    } else {
-      formatted = executePrettier(editor, text);
-    }
+const executePrettierOrIntegration = async (editor: TextEditor, text: string, cursorOffset: number) => {
+  const nextCursorOffset: number = cursorOffset;
+  let nextText;
 
-    return formatted;
-  } catch (error) {
-    return error;
+  if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
+    // TODO: add support for cursor position - https://github.com/hugomrdias/prettier-stylelint/issues/13
+    nextText = await executePrettierStylelint(editor, text);
+  } else if (shouldUseEslint()) {
+    // TODO: add support for cursor position - https://github.com/prettier/prettier-eslint/issues/164
+    nextText = executePrettierEslint(editor, text);
+  } else {
+    return executePrettierWithCursor(editor, text, cursorOffset);
   }
+
+  return { formatted: nextText, cursorOffset: nextCursorOffset };
 };
 
 const executePrettierOnBufferRange = async (
   editor: TextEditor,
   bufferRange: Range,
   options?: { setTextViaDiff?: boolean },
 ) => {
-  const cursorPositionPriorToFormat = editor.getCursorScreenPosition();
-  const textToTransform = editor.getTextInBufferRange(bufferRange);
-  const transformed = await executePrettierOrIntegration(editor, textToTransform);
+  // grab cursor position and file contents
+  const currentBuffer: TextBuffer = editor.getBuffer();
+  const cursorPosition: Point = editor.getCursorBufferPosition();
+  const source: string = editor.getTextInBufferRange(bufferRange);
```

Comment:
let's keep as `textToTransform`, `source` is a bit nebulous

Reply:
I was naming based off the Prettier API documentation. I can switch it back if you prefer. Personally, I prefer to stick with terms that exist in the documentation people would be looking at when working with external systems.

Reply:
I see both sides on this one, IMO `source` isn't a good name in the context of what's going on here, but yes it is helpful to be consistent with naming.